src/python.rs

```diff
@@ -102,16 +136,15 @@ impl<'j> PythonParser<'j> {
                         panic!("PyDict_SetItem failed")
                     }
                 };
-
-                if let Some(first_key) = self.parser.object_first::<StringDecoder>(&mut self.tape)? {
+                if let Some(first_key) = tri!(self.parser.object_first::<StringDecoder>(&mut self.tape), dict) {
```

Comment:
Similar thought as with lists, if you like the idea.

Reply:
done.